plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsService.php

```diff
@@ -632,35 +741,160 @@ public function reset_onboarding( string $from = '', string $source = '' ): arra
-			array(
-				'from'   => ! empty( $from ) ? esc_attr( $from ) : self::FROM_PAYMENT_SETTINGS,
-				'source' => ! empty( $source ) ? esc_attr( $source ) : self::FROM_PAYMENT_SETTINGS,
-			)
-		);
+		$this->check_if_onboarding_action_is_acceptable();
+
+		// Lock the onboarding to prevent concurrent actions.
+		$this->set_onboarding_lock();
+
+		try {
+			// Call the WooPayments API to disable the test account and prepare for the switch to live.
+			$response = $this->proxy->call_static(
+				Utils::class,
+				'rest_endpoint_post_request',
+				'/wc/v3/payments/onboarding/test_drive_account/disable',
+				array(
+					'from'   => ! empty( $from ) ? esc_attr( $from ) : self::FROM_PAYMENT_SETTINGS,
+					'source' => ! empty( $source ) ? esc_attr( $source ) : self::FROM_PAYMENT_SETTINGS,
+				)
+			);
+		} catch ( Exception $e ) {
+			// Catch any exceptions to allow for proper error handling and onboarding unlock.
+			$response = new WP_Error(
+				'woocommerce_woopayments_onboarding_client_api_exception',
+				esc_html__( 'An unexpected error happened while disabling the test account.', 'woocommerce' ),
+				array(
+					'code'    => $e->getCode(),
+					'message' => $e->getMessage(),
+					'trace'   => $e->getTrace(),
+				)
+			);
+		}
+
+		// Unlock the onboarding after the API call finished or errored.
+		$this->clear_onboarding_lock();
 
 		if ( is_wp_error( $response ) ) {
-			throw new Exception( esc_html( $response->get_error_message() ) );
+			throw new ApiException(
+				'woocommerce_woopayments_onboarding_client_api_error',
+				esc_html( $response->get_error_message() ),
+				(int) WP_Http::FAILED_DEPENDENCY,
+				map_deep( (array) $response->get_error_data(), 'esc_html' )
+			);
 		}
 
 		if ( ! is_array( $response ) || empty( $response['success'] ) ) {
-			throw new Exception( esc_html__( 'Failed to disable test account.', 'woocommerce' ) );
+			throw new ApiException(
+				'woocommerce_woopayments_onboarding_client_api_error',
+				esc_html__( 'Failed to disable the test account.', 'woocommerce' ),
+				(int) WP_Http::FAILED_DEPENDENCY
+			);
 		}
 
 		// For sanity, make sure the payment methods step is marked as completed.
 		// This is to avoid the user being prompted to set up payment methods again.
 		$this->set_onboarding_step_completed( self::ONBOARDING_STEP_PAYMENT_METHODS, $location );
+		// For sanity, make sure the test account step is marked as completed.
+		// After disabling a test account, the user should be prompted to set up a live account.
+		$this->set_onboarding_step_completed( self::ONBOARDING_STEP_TEST_ACCOUNT, $location );
 
 		return $response;
 	}
 
+	/**
+	 * Check if an onboarding action should be allowed to be processed.
+	 *
+	 * @return void
+	 * @throws ApiException If the extension is not active or onboarding is locked.
+	 */
+	private function check_if_onboarding_action_is_acceptable() {
+		// If the WooPayments plugin is not active, we can't do anything.
+		if ( ! $this->is_extension_active() ) {
```

Comment:
Should we also check the version of WooPayments here? Or are we leaving that to the UI-only?

Reply:
Good call, Ahmed! I don't see a reason not to enforce it on the backend. I made [the changes](https://github.com/woocommerce/woocommerce/pull/57591/commits/a993c823756baf05b494a98305304a6a967d5250).